core/templates/pages/admin-page/roles-tab/roles-and-actions-visualizer.component.html

```diff
@@ -0,0 +1,53 @@
+<style>
+  .oppia-roles-container {
+    display: flex;
+    flex-wrap: wrap;
+    justify-content: space-between;
+    align-items: center;
+    background-color: rgba(204, 204, 204, 0.438);
+    align-content: space-around;
+    padding: 10px;
+  }
+  .oppia-role-container {
+    margin: 10px 0;
+    text-align: justify;
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    min-width: 100px;
+  }
+  .oppia-role-avatar {
+    display: block;
+    margin: 0 auto;
+    width: 50px;
+  }
+  .oppia-role-name-lable {
+    font-size: 12px;
+    margin-top: 10px;
+  }
+  .oppia-actions-container {
+    min-height: 200px;
+    margin-top: 50px;
+  }
+  .oppia-selected-role {
+    background: #aaa;
+    border-radius: 8px;
+  }
+</style>
+
+<div class="oppia-roles-container">
+  <div class="oppia-role-container" [ngClass]="{'oppia-selected-role': activeRole === role}" *ngFor="let role of roles" (click)="setActiveRole(role)">
+    <img class="oppia-role-avatar" [src]="avatarPictureUrl" alt="User with {{ role }} role">
+    <span class="oppia-role-name-lable"><strong>{{ role }}</strong></span>
+  </div>
+</div>
+
+<div class="oppia-actions-container">
+  <strong>Actions allocated to {{ activeRole }}:</strong>
+  <ul>
+    <li *ngIf="!roleToActions[activeRole].length">
```

Comment:
You should check first wheather `roleToActions[activeRole]` is defined or not.
The template is compiled and rendered before triggering ngOnInit. So, it may cause console errors regarding this.

I'm already getting them sometimes.
![Screenshot from 2021-06-07 15-19-49](https://user-images.githubusercontent.com/24855641/120996776-428c7f80-c7a4-11eb-8216-f665fad7b88a.png)

Reply:
You can do something like
```html
<ng-container *ngIf="roleToActions[activeRole]">
   <li *ngIf="!roleToActions[activeRole].length">
       <i class="oppia-no-actions-message">{{ activeRole }} cannot perform any actions.</i>
   <li>
</ng-container>
```

Reply:
`*ngIf="!roleToActions[activeRole]?.length"` this fix might help here

Reply:
Created PR #13018 as a fix.

Reply:
@ashutoshc8101 Thanks for creating the fix (I was not sure for this case to happen as I have blocked the visualizer directive to happen until rolesToActions is defined., but I agree that the switching tab can create this issue)!